src/transformers/models/maskformer/feature_extraction_maskformer.py

```diff
@@ -0,0 +1,568 @@
+        elif isinstance(images, (list, tuple)):
+            if len(images) == 0 or isinstance(images[0], (Image.Image, np.ndarray)) or is_torch_tensor(images[0]):
+                valid_images = True
+
+        if not valid_images:
+            raise ValueError(
+                "Images must of type `PIL.Image.Image`, `np.ndarray` or `torch.Tensor` (single example), "
+                "`List[PIL.Image.Image]`, `List[np.ndarray]` or `List[torch.Tensor]` (batch of examples)."
+            )
+
+        is_batched = bool(
+            isinstance(images, (list, tuple))
+            and (isinstance(images[0], (Image.Image, np.ndarray)) or is_torch_tensor(images[0]))
+        )
+
+        if not is_batched:
+            images = [images]
+            if annotations is not None:
+                annotations = [annotations]
+
+        # Check that annotations has a valid type
+        if annotations is not None:
+            valid_annotations = type(annotations) is list and "masks" in annotations[0] and "labels" in annotations[0]
+            if not valid_annotations:
+                raise ValueError(
+                    """
+                    Annotations must of type `Dict` (single image) or `List[Dict]` (batch of images). The annotations
+                    must be numpy arrays in the following format: { "masks" : the target mask, with shape [C,H,W],
+                    "labels" : the target labels, with shape [C]}
+                    """
+                )
+
+        # transformations (resizing + normalization)
+        if self.do_resize and self.size is not None:
+            if annotations is not None:
+                for idx, (image, target) in enumerate(zip(images, annotations)):
+                    image, target = self._resize(image=image, target=target, size=self.size, max_size=self.max_size)
+                    images[idx] = image
+                    annotations[idx] = target
+            else:
+                for idx, image in enumerate(images):
+                    images[idx] = self._resize(image=image, target=None, size=self.size, max_size=self.max_size)[0]
+
+        if self.do_normalize:
+            if annotations is not None:
+                for idx, (image, target) in enumerate(zip(images, annotations)):
+                    image, target = self._normalize(
+                        image=image, mean=self.image_mean, std=self.image_std, target=target
+                    )
+                    images[idx] = image
+                    annotations[idx] = target
+            else:
+                images = [
+                    self._normalize(image=image, mean=self.image_mean, std=self.image_std)[0] for image in images
+                ]
+        # NOTE I will be always forced to pad them them since they have to be stacked in the batch dim
+        encoded_inputs = self.encode_inputs(
+            images, annotations, pad_and_return_pixel_mask, return_tensors=return_tensors
+        )
+
+        if annotations is not None:
```

Comment:
I know this comes from DETR's feature extractor, but shouldn't that be a check on `return_tensors` instead of `annotations`?

I'm not sure I understand this part, could you explain it to me? We're only raising errors when `annotations` is passed, but otherwise we're correctly returning TensorFlow tensors if asked. This differs from DETR which checks the same but also converts tensors to PyTorch in this if statement, which isn't the case here.

Reply:
You are correct, nice catch. Removed the `if annotations is not None` and added test.